d3-scripts/src/d3_scripts/json_tools.py

```diff
@@ -27,11 +32,28 @@ def is_json_unchanged(file_name: str, claim: dict):
     try:
         with open(file_name) as f:
             json_data = json.load(f)
-        return json_data == claim
+        return is_json_same_as_claim(json_data, claim)
     except FileNotFoundError:
         return False
 
 
+def is_json_same_as_claim(json_data, claim):
+    def compare_properties(key, flat_json_data, flat_claim_data):
+        if key == "credentialSubject.behaviour":
+            return (flat_claim_data[key] == flat_json_data["credentialSubject.behaviour.id"] or
+                    flat_claim_data[key] == flat_json_data["credentialSubject.behaviour.name"])
+        else:
+            return flat_json_data[key] == flat_claim_data[key]
```

Comment:
Does this need to be defined inside `is_json_same_as_claim`? It looks like all the variables are captured by the function args